account-kit/core/src/store/store.ts

```diff
@@ -48,14 +48,17 @@ export const createAccountKitStore = (
                     ),
                   };
                 }
+
                 return bigintMapReplacer(key, value);
               },
               reviver: (key, value) => {
                 if (key === "bundlerClient") {
-                  const { connection } = value as { connection: Connection };
+                  const {
```

Comment:
oof this function was hard to follow on what the merge behavior was exactly trying to do. We might want to break this up a bit and add some comments. This is the type of function that inline stuff doesn't really do us any favors haha. It might make testing above a bit easier too. Not a blocker given it's out of scope of this PR but the type of thing that will help us maintain this code base longer term.

Reply:
I felt this big time as well so I refactored it: in https://github.com/alchemyplatform/aa-sdk/pull/997